### DISCUSSION TITLE
Stream file is showed as plain/text (instead of plainhtml)

### DISCUSSION THREAD

**Discussion by ernitron** [miguelgrinberg/microdot#194] [Q&A]
First of all KUDOS to Miguel for this awesome piece of art. I have it running on ESP8266 (frozen code) and ESP32 and it shines!

I manage to send file (I have my own template that substitutes patterns like {{contents}} found in file (code is below). I use stream with generating function that read the file, substitute the content and yields file line by line. And it works!
However, whatever I do to convince Microdot that is a plain/html the browser don't get and shows as plain/text. If I refresh page is rendered correctly. What am I doing wrong here?

Code snippet:

```
def rendertemplate(template='template.html', **key):
    from Microdot import Response
     Response.default_content_type = 'text/html'
     try:
         with open(template, 'r') as f:
             for line in f.readlines():
                 for k in key:
                     tag = "{{%s}}" % k # e.g builds tag: {{content}}
                     if tag in line:
                         line = line.replace(tag, key[k])
                 yield line
     except Exception as e:
         yield str(e)
```

**Comment by ernitron** [comment]
Maybe to complete the issue I should also add some more snippets:

```
@app.get('/helloworld')
async def helloworld(request):
    content = 'Hello, World!'
    return rendertemplate(content=content)
```

Template is as simple as:

```
<!doctype html>
<html lang="en">
<head>
</head>
<body>
{{content}}
</body></html>
```

**Reply by miguelgrinberg** [reply]
This line:

    Response.default_content_type = 'text/html'

Should be in the global scope, not inside your template rendering function. If you want the response to be HTML only for some endpoints but not all, then add a `Content-Type` header explicitly in the endpoints that you want.
